src/analysis_and_optimization/Mir_utils.ml

```diff
@@ -426,6 +426,388 @@ let rec update_expr_ad_levels autodiffable_variables
 and update_idx_ad_levels autodiffable_variables =
   Index.map (update_expr_ad_levels autodiffable_variables)
 
+(**
+ * Modify an index's inner expressions with `op`
+ * @param op a functor returning an expression
+ * @param ind the Index.t to modify
+ *)
+let mod_index op ind =
+  match ind with
+  | Index.All -> Index.All
+  | Single ind_expr -> Single (op ind_expr)
+  | Upfrom ind_expr -> Upfrom (op ind_expr)
+  | Between (expr_top, expr_bottom) -> Between (op expr_top, op expr_bottom)
+  | MultiIndex exprs -> MultiIndex (op exprs)
+
+(**
+ * Apply an op returning true/false to an index 
+ * @param op a functor returning a boolean
+ * @param ind the Index.t to modify
+ *)
+let search_index op ind =
```

Comment:
This is usually called an `any` function by convention (it'd be an `all` with && instead of ||)